internal/controller/storageclient_controller.go

```diff
@@ -368,42 +372,57 @@ func (r *storageClientReconcile) reconcilePhases() (ctrl.Result, error) {
 
 	r.storageClient.Status.InMaintenanceMode = storageClientResponse.MaintenanceMode
 
+	resourceMapping := gvkToObjectKeys{}
 	for _, kubeResource := range storageClientResponse.KubeResources {
 		var err error
 		typeMeta := &metav1.TypeMeta{}
 		if err := json.Unmarshal(kubeResource, typeMeta); err != nil {
 			return reconcile.Result{}, fmt.Errorf("failed to unmarshal the type for the Object: %w", err)
 		}
 
+		var resourceObj client.Object = nil
+		shouldReplace := false
 		// Create the received resources, if necessary.
 		switch typeMeta.GroupVersionKind() {
 		case quotav1.SchemeGroupVersion.WithKind("ClusterResourceQuota"):
-			err = r.reconcileResource(&quotav1.ClusterResourceQuota{}, kubeResource, false)
+			resourceObj = &quotav1.ClusterResourceQuota{}
+			shouldReplace = false
```

Comment:
The default is already false, there is no point to set it explicitly for false values
```suggestion
```

Reply:
Yes, I thought of it but still made it to be explicit incase the outer code moves or changes meaning.

Anyways, since we already know there is another pr removing this, will make the change.

Reply:
done.